pkg/controller/service/service_controller.go

```diff
@@ -248,7 +248,12 @@ func (s *ServiceController) processServiceUpdate(cachedService *cachedService, s
 	cachedService.state = service
 	err, retry := s.createLoadBalancerIfNeeded(key, service)
 	if err != nil {
+		eventType := "CreatingLoadBalancerFailed"
 		message := "Error creating load balancer"
+		if !wantsLoadBalancer(service) {
```

Comment:
Why not calling this before `createLoadBalancerIfNeeded`?

If a LB is not needed, there is no need to create it, right?

Reply:
As I described above, func `createLoadBalancerIfNeeded` not only in charge of LB creation, but also try to clean up potential orphaned load balancer if the service does not need it any more. Func name `createLoadBalancerIfNeeded` seems a little  confusing somehow.
Another alternative is to break func `createLoadBalancerIfNeeded` into two parts. However, it might be overengineering of making such a big change just to modify the event message.
Would like to hear the opinions of guys from sig network.

Reply:
> Another alternative is to break the content of func createLoadBalancerIfNeeded into two parts, according to the fact a service need a LB or not. However, it might be overengineering of making such a big change just to modify the event message.

Agree `createLoadBalancerIfNeeded` is not a great name, but let's keep this for now if it doesn't block the event change.